From ec85ad89f351128b69e45e92f7338cea542a186b Mon Sep 17 00:00:00 2001
From: Konstantin Porotchkin <kostap@marvell.com>
Date: Sun, 4 Apr 2021 13:46:15 +0300
Subject: [PATCH 1405/1921] arch/arm64/boot/dts: cp110 - add msi i2c device

Add missing MSS i2c device to CP11x device tree

Signed-off-by: Konstantin Porotchkin <kostap@marvell.com>
Change-Id: I97843344546117a4f70334969b5359acb343d6a4
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/49158
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Nadav Haklai <Nadav.Haklai@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-cp11x.dtsi | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi b/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
index 935fa55fa1bf..249dbe764559 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp11x.dtsi
@@ -425,6 +425,18 @@
 			status = "disabled";
 		};
 
+		CP11X_LABEL(mss_i2c): i2c@211000 {
+			compatible = "marvell,mv78230-i2c";
+			reg = <0x211000 0x20>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+			interrupts = <0 IRQ_TYPE_LEVEL_HIGH>;
+			clock-names = "core", "reg";
+			clocks = <&CP11X_LABEL(clk) 1 21>,
+				 <&CP11X_LABEL(clk) 1 17>;
+			status = "disabled";
+		};
+
 		CP11X_LABEL(uart0): serial@702000 {
 			compatible = "snps,dw-apb-uart";
 			reg = <0x702000 0x100>;
-- 
2.31.1

